test cases/windows/24 diasdk copy dll/meson.build

```diff
@@ -0,0 +1,21 @@
+project('diatest', 'cpp')
+
+if host_machine.system() != 'windows'
+  error('MESON_SKIP_TEST: unsupported platform')
+endif
+cpp = meson.get_compiler('cpp', native: false)
+is_msvc_clang = cpp.get_id() == 'clang' and cpp.get_define('_MSC_VER') != ''
+if not ['msvc', 'clang-cl'].contains(cpp.get_id()) and not is_msvc_clang
+  error('MESON_SKIP_TEST: unsupported compiler')
+endif
+
+dia = dependency('diasdk', required: true)
+dia_dll_name = dia.get_variable(internal: 'dll')
```

Comment:
You don't need "internal" here. The get_variable() method accepts a generic variable name, and a type-specific one. The latter is intended to be useful for cases where e.g. the same information is available via both pkg-config and cmake but the name of the pkg-config variable is different from the name of the cmake variable.

(e.g. because cmake loves capitalized variable names.)

It is therefore necessary to pass a different name into get_variable() depending on which type of dependency was found -- or use `dep.get_variable(pkgconfig: 'foo', cmake: 'FOO')`